From 9ba0694574761947d1e414bcc4ed1a903a556c86 Mon Sep 17 00:00:00 2001
From: Assam Boudjelthia <assam.boudjelthia@qt.io>
Date: Tue, 8 Sep 2020 17:47:28 +0300
Subject: Android: don't use avx and avx2 when building for Android x86_64

The Android build for x86_64 fails with CMake. It should also fail with
qmake builds but the current multi-ABI build is testing the feature for
arm64-v8a only.

In [1], it's stated that AVX and AVX2 are not supported on x86_64 ABI.

[1] https://developer.android.com/ndk/guides/abis#86-64

Task-number: QTBUG-85982
Change-Id: I20c9366bdee2a13ca554ec67f5d0c041bd068b91
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit ff1a44be33f4bc05d502a2ca49171e0408992f61)
---
 configure.json | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git x/qtbase/configure.json y/qtbase/configure.json
index c273aeeef1..4a7c50f684 100644
--- x/qtbase/configure.json
+++ y/qtbase/configure.json
@@ -1098,7 +1098,7 @@
         },
         "avx": {
             "label": "AVX",
-            "condition": "features.sse4_2 && tests.avx",
+            "condition": "features.sse4_2 && tests.avx && (!config.android || !arch.x86_64)",
             "output": [
                 "privateConfig",
                 { "type": "define", "name": "QT_COMPILER_SUPPORTS_AVX", "value": 1 }
@@ -1114,7 +1114,7 @@
         },
         "avx2": {
             "label": "AVX2",
-            "condition": "features.avx && tests.avx2",
+            "condition": "features.avx && tests.avx2 && (!config.android || !arch.x86_64)",
             "output": [
                 "privateConfig",
                 "privateFeature",
-- 
2.33.0

